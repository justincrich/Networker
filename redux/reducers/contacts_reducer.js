import * as ActionTypes from '../actiontypes/actiontypes';
// import {REHYDRATE} from 'redux-persist/constants';

const initialState={

    contacts:[],
    requestedId:null,
    selectedContact:{},
    status:'ready',
    error:null,
}

export default function ContactsReducer(state=initialState,action){
    switch(action.type){
        case ActionTypes.CONTACT_REQUEST_ONE_BEGIN:{
            return{
                ...state,
                requestedId:action.id,
                status:'loading'
            }
        }
        case ActionTypes.CONTACT_REQUEST_ONE_RESOLVE:{
            return{
                ...state,
                status:'ready',
                requestedId:null,
                selectedContact:action.contact
            }
        }
        case ActionTypes.CONTACTLIST_BEGIN_GET_ALL:{
            return{
                ...state,
                status:'loading'
            }
        }
        case ActionTypes.CONTACTLIST_RECEIVE_ALL:{
            return{
                ...state,
                status:'complete',
                contacts:action.contacts
            }
        }
        case ActionTypes.CONTACT_REQUEST:{
            let selection = state.contacts[action.id];
            return {
                ...state,
                selectedContact:selection
            }
        }
        case ActionTypes.CONTACT_BEGIN_CREATE:{
            return {
                ...state,
                status:'loading'
            }
        }
        case ActionTypes.CONTACT_RESOLVE_CREATE:{
            return {
                ...state,
                status:'complete'
            }
        }
        case ActionTypes.CONTACT_THROW_ERROR:{
            return {
                ...state,
                error:action.error,
                status:'error'
            }
        }
        case ActionTypes.CONTACT_RESET_STATUS:{
            return{
                ...state,
                status:'ready'
            }
        }
        default:
            return state;
    }
}