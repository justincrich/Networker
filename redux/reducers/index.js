// import { combineReducers } from 'redux';
// import ContactsDataReducer from './contacts_reducer';

// export default combineReducers({
//     contacts:ContactsDataReducer
// });