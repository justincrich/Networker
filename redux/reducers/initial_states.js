export const contactsInitialState={
    
        contacts:[],
        requestedId:null,
        requestDeleteId:null,
        selectedContact:undefined,
        status:'ready',
        error:null,
    }