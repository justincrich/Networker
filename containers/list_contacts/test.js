
import React from 'react';
import {shallow} from 'enzyme';
import ListContactContainer from './list_contacts_container';
const navMock = { 
                    navigate: jest.fn(),
                    state:{
                        routeName:"ListContacts"
                    }
                    
                    };

// //redux objects
import configureStore from 'redux-mock-store';
const thunk = require('redux-thunk').default;
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
import {contactsInitialState} from '../../redux/reducers/initial_states';


// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';


describe('Contact List Container',()=>{
    beforeAll(()=>{
        contactsInitialState.contacts.push(
            
            {
            id: 1511821971418,
            firstName: 'Justin',
            lastName: 'Rich',
            pictureUri: 'file:///storage/emulated/0/Android/data/com.networker/files/Pictures/image-fc525719-52d5-489a-98ac-fbd3118385c6.jpg',
            email: 'J@j.com',
            phoneNumber: '(555) 555-5555',
            notes: 'Hi Justin',
            jobTitle: 'Developer',
            company: 'Hackerpug'
            },
            {
                id: 1511821971417,
                firstName: 'Sam',
                lastName: 'Joe',
                pictureUri: 'file:///storage/emulated/0/Android/data/com.networker/files/Pictures/image-fc525719-52d5-489a-98ac-fbd3118385c6.jpg',
                email: 's@s.com',
                phoneNumber: '(555) 555-5555',
                notes: 'Hi Sam',
                jobTitle: 'Product Manager',
                company: 'Hackerpug'
            }
        )
    })
    it('renders correctly', () => {
        const wrapper = shallow(
          <ListContactContainer/>,
          {context:{store:mockStore(contactsInitialState)}}
        );
        wrapper.setProps({navigation:navMock})
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('gets all contacts in database',()=>{

    });
})



// describe('Contact List Container',()=>{
//     it('renders correctly', () => {
//         const tree = renderer.create(
//           <ListContactContainer store={store} navigation={navigation}/>
//         ).toJSON();
//         expect(tree).toMatchSnapshot();
//     });
// })
// import * as ContactActionTypes from '../../redux/actiontypes/actiontypes';
// import * as ContactActions from '../../redux/actions/contacts_actions';
// import * as ContactReducer from '../../redux/reducers/contacts_reducer';