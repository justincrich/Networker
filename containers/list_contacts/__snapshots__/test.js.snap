// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact List Container renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListContactContainer
    Actions={
      Object {
        "beginContactCreate": [Function],
        "beginContactDelete": [Function],
        "beginContactUpdate": [Function],
        "beginGetAllContacts": [Function],
        "beginGetContact": [Function],
        "clearSelectedContact": [Function],
        "getAllContacts": [Function],
        "getContact": [Function],
        "receiveAllContacts": [Function],
        "requestContactCreate": [Function],
        "requestDeleteContact": [Function],
        "requestUpdateContact": [Function],
        "resetContactStatus": [Function],
        "resolveContactCreate": [Function],
        "resolveContactDelete": [Function],
        "resolveContactUpdate": [Function],
        "resolveGetContact": [Function],
        "throwContactError": [Function],
      }
    }
    data={
      Object {
        "contacts": Array [
          Object {
            "company": "Hackerpug",
            "email": "J@j.com",
            "firstName": "Justin",
            "id": 1511821971418,
            "jobTitle": "Developer",
            "lastName": "Rich",
            "notes": "Hi Justin",
            "phoneNumber": "(555) 555-5555",
            "pictureUri": "file:///storage/emulated/0/Android/data/com.networker/files/Pictures/image-fc525719-52d5-489a-98ac-fbd3118385c6.jpg",
          },
          Object {
            "company": "Hackerpug",
            "email": "s@s.com",
            "firstName": "Sam",
            "id": 1511821971417,
            "jobTitle": "Product Manager",
            "lastName": "Joe",
            "notes": "Hi Sam",
            "phoneNumber": "(555) 555-5555",
            "pictureUri": "file:///storage/emulated/0/Android/data/com.networker/files/Pictures/image-fc525719-52d5-489a-98ac-fbd3118385c6.jpg",
          },
        ],
        "error": null,
        "requestDeleteId": null,
        "requestedId": null,
        "selectedContact": undefined,
        "status": "ready",
      }
    }
    navigation={
      Object {
        "navigate": [Function],
        "state": Object {
          "routeName": "ListContacts",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Header
          navigate={[Function]}
          routeName="ListContacts"
          type="search"
        />,
        <ActionButton
          name="md-add"
          onPress={[Function]}
          type="ionicon"
        />,
        <ListContactComponent
          contacts={
            Array [
              Object {
                "company": "Hackerpug",
                "email": "J@j.com",
                "firstName": "Justin",
                "id": 1511821971418,
                "jobTitle": "Developer",
                "lastName": "Rich",
                "notes": "Hi Justin",
                "phoneNumber": "(555) 555-5555",
                "pictureUri": "file:///storage/emulated/0/Android/data/com.networker/files/Pictures/image-fc525719-52d5-489a-98ac-fbd3118385c6.jpg",
              },
              Object {
                "company": "Hackerpug",
                "email": "s@s.com",
                "firstName": "Sam",
                "id": 1511821971417,
                "jobTitle": "Product Manager",
                "lastName": "Joe",
                "notes": "Hi Sam",
                "phoneNumber": "(555) 555-5555",
                "pictureUri": "file:///storage/emulated/0/Android/data/com.networker/files/Pictures/image-fc525719-52d5-489a-98ac-fbd3118385c6.jpg",
              },
            ]
          }
          openContact={[Function]}
        />,
      ],
      "style": Object {
        "flex": 1,
        "position": "relative",
        "top": 0,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "navigate": [Function],
          "routeName": "ListContacts",
          "type": "search",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": "md-add",
          "onPress": [Function],
          "type": "ionicon",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "contacts": Array [
            Object {
              "company": "Hackerpug",
              "email": "J@j.com",
              "firstName": "Justin",
              "id": 1511821971418,
              "jobTitle": "Developer",
              "lastName": "Rich",
              "notes": "Hi Justin",
              "phoneNumber": "(555) 555-5555",
              "pictureUri": "file:///storage/emulated/0/Android/data/com.networker/files/Pictures/image-fc525719-52d5-489a-98ac-fbd3118385c6.jpg",
            },
            Object {
              "company": "Hackerpug",
              "email": "s@s.com",
              "firstName": "Sam",
              "id": 1511821971417,
              "jobTitle": "Product Manager",
              "lastName": "Joe",
              "notes": "Hi Sam",
              "phoneNumber": "(555) 555-5555",
              "pictureUri": "file:///storage/emulated/0/Android/data/com.networker/files/Pictures/image-fc525719-52d5-489a-98ac-fbd3118385c6.jpg",
            },
          ],
          "openContact": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Header
            navigate={[Function]}
            routeName="ListContacts"
            type="search"
          />,
          <ActionButton
            name="md-add"
            onPress={[Function]}
            type="ionicon"
          />,
          <ListContactComponent
            contacts={
              Array [
                Object {
                  "company": "Hackerpug",
                  "email": "J@j.com",
                  "firstName": "Justin",
                  "id": 1511821971418,
                  "jobTitle": "Developer",
                  "lastName": "Rich",
                  "notes": "Hi Justin",
                  "phoneNumber": "(555) 555-5555",
                  "pictureUri": "file:///storage/emulated/0/Android/data/com.networker/files/Pictures/image-fc525719-52d5-489a-98ac-fbd3118385c6.jpg",
                },
                Object {
                  "company": "Hackerpug",
                  "email": "s@s.com",
                  "firstName": "Sam",
                  "id": 1511821971417,
                  "jobTitle": "Product Manager",
                  "lastName": "Joe",
                  "notes": "Hi Sam",
                  "phoneNumber": "(555) 555-5555",
                  "pictureUri": "file:///storage/emulated/0/Android/data/com.networker/files/Pictures/image-fc525719-52d5-489a-98ac-fbd3118385c6.jpg",
                },
              ]
            }
            openContact={[Function]}
          />,
        ],
        "style": Object {
          "flex": 1,
          "position": "relative",
          "top": 0,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "navigate": [Function],
            "routeName": "ListContacts",
            "type": "search",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "md-add",
            "onPress": [Function],
            "type": "ionicon",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "contacts": Array [
              Object {
                "company": "Hackerpug",
                "email": "J@j.com",
                "firstName": "Justin",
                "id": 1511821971418,
                "jobTitle": "Developer",
                "lastName": "Rich",
                "notes": "Hi Justin",
                "phoneNumber": "(555) 555-5555",
                "pictureUri": "file:///storage/emulated/0/Android/data/com.networker/files/Pictures/image-fc525719-52d5-489a-98ac-fbd3118385c6.jpg",
              },
              Object {
                "company": "Hackerpug",
                "email": "s@s.com",
                "firstName": "Sam",
                "id": 1511821971417,
                "jobTitle": "Product Manager",
                "lastName": "Joe",
                "notes": "Hi Sam",
                "phoneNumber": "(555) 555-5555",
                "pictureUri": "file:///storage/emulated/0/Android/data/com.networker/files/Pictures/image-fc525719-52d5-489a-98ac-fbd3118385c6.jpg",
              },
            ],
            "openContact": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
